ssl/ssl_lib.c

```diff
@@ -4074,45 +4070,138 @@ int SSL_CTX_set_ct_validation_callback(SSL_CTX *ctx, ct_validation_cb callback,
             TLSEXT_TYPE_signed_certificate_timestamp)) {
         SSLerr(SSL_F_SSL_CTX_SET_CT_VALIDATION_CALLBACK,
                SSL_R_CUSTOM_EXT_HANDLER_ALREADY_INSTALLED);
-        goto err;
+        return 0;
     }
 
     ctx->ct_validation_callback = callback;
     ctx->ct_validation_callback_arg = arg;
-    ret = 1;
-err:
-    return ret;
+    return 1;
 }
 
-ct_validation_cb SSL_get_ct_validation_callback(const SSL *s)
+int SSL_ct_is_enabled(const SSL *s)
 {
-    return s->ct_validation_callback;
+    return s->ct_validation_callback != NULL;
 }
 
-ct_validation_cb SSL_CTX_get_ct_validation_callback(const SSL_CTX *ctx)
+int SSL_CTX_ct_is_enabled(const SSL_CTX *ctx)
 {
-    return ctx->ct_validation_callback;
+    return ctx->ct_validation_callback != NULL;
+}
+
+static void *ssl_ct_arg(int vtype)
```

Comment:
Oh I wasn't proposing that you'd pass the enum value in as a void*. You'd change the void* to the enum type (or int) since `ssl_ct_cb` would no longer have to conform to the callback signature, as it wouldn't be called directly. You'd want to rename it too really. You seem to get the same flexibility, but avoid needing this `ssl_ct_arg`  function, which I'm not keen on. Needing a dedicated function to map each integer value of the enum to a pointer to that enum value, which is then passed as a void*, just seems messy in comparison.